frontend/src/pages/main/profile/components/profileInformation/ProfileInformation.tsx

```diff
@@ -0,0 +1,92 @@
+import {Notifications} from "../../../../../common/constansts/notifications";
+import {Titles} from "../../../../../common/constansts/titles";
+import CardTitle from "../../../../../components/typography/cardTitle/CardTitle";
+
+const initialFieldsValidation: FieldsValidation = {
+    isEmailValid: false,
+    isPasswordValid: true,
+    isNameValid: false
+};
+
+const initialUserModification: UserModification = {
+    name: "",
+    email: "",
+    password: ""
+}
+
+const ProfileInformation = () => {
+
+    const [userModification, setUserModification] = useState<UserModification>(initialUserModification);
+    const [fieldsValidation, setFieldsValidation] = useState<FieldsValidation>(initialFieldsValidation);
+    const [isValid, setIsValid] = useState<boolean>(false);
+    const loggedUser = useAppSelector(selectLoggedUser);
+    const dispatch = useAppDispatch();
+
+    const updateEmail = (email: string, isEmailValid: boolean) => {
+        setUserModification({...userModification, email});
+        setFieldsValidation({...fieldsValidation, isEmailValid});
+    }
+
+    const updateName = (name: string, isNameValid: boolean) => {
+        setUserModification({...userModification, name});
+        setFieldsValidation({...fieldsValidation, isNameValid});
+    }
+
+    const updatePassword = (password: string, isPasswordValid: boolean) => {
+        setUserModification({...userModification, password});
+        if (password === "") {
+            const validateEmptyPassword = !isPasswordValid;
+            setFieldsValidation({...fieldsValidation, isPasswordValid: validateEmptyPassword});
+        } else {
+            setFieldsValidation({...fieldsValidation, isPasswordValid});
+        }
+    }
+
+    const validateData = () => {
+        const fieldsValid = Object.values(fieldsValidation).every(field => field);
+        setIsValid(fieldsValid);
+    };
+
+    const handleConfirm = async () => {
+        const updatedUser = await updateUserService(loggedUser.id, loggedUser.userType, userModification);
+        if (!updatedUser) return;
+        dispatch(showNotification({
+            show: true,
+            type: NotificationType.SUCCESS,
+            message: Notifications.SUCCESSFUL_UPDATE_USER_DATA
+        }));
+        dispatch(logOut());
+    }
+
+    useEffect(() => {
+        validateData();
+    }, [fieldsValidation]);
+
+    return (
+        <Card className="card-background card-payment">
+            <CardContent className="card-confirm-content">
+                <CardTitle title={Titles.INFORMATION}/>
+                <EmailField defaultEmail={userModification.email} updateEmailFields={updateEmail}/>
+                <NameField defaultName={userModification.name} updateNameFields={updateName}/>
+                <PasswordField updatePasswordFields={updatePassword}/>
+            </CardContent>
+            <ConfirmationButton handleClick={handleConfirm} buttonName={Buttons.CONFIRM} isDisabled={!isValid}/>
```

Comment:
buttonName should be optional parameter with default value `Buttons.CONFIRM`. Same for cancel button.

Reply:
```typescript
export interface ConfirmationButtonProps {
    buttonName?: string;
    handleClick: () => void;
    isDisabled: boolean;
}


const ConfirmationButton: React.FC<ConfirmationButtonProps> = ({handleClick, buttonName, isDisabled}) => {

    return (
        <Button
            onClick={handleClick}
            variant="contained"
            className="button"
            disabled={isDisabled}
        >
            {buttonName ? buttonName : Buttons.CONFIRM}
        </Button>
    );
};
```

Reply:
Also you can reuse button props:
```typescript
export interface ButtonProps {
    buttonName?: string;
    handleClick: () => void;
}

export interface ConfirmationButtonProps extends ButtonProps{
    isDisabled: boolean;
}
```